### PR TITLE
[Qt] Fix PIV balances on overview page

### LINKED ISSUE #654: Locked balance being double counted
### Describe the issue
Locked PIV balances are being "double counted" on the overview tab such that they cause the "Available" and "Total" values to be incorrect/over calculated.

#### If so, please list the steps to reproduce below:
1. Run GUI wallet
2. Have locked coins (MN collateral or otherwise locked UTXO)

### Expected behavior
Locked coins should not be reflected as "Available", and the calculation of "Total" represented as `available + pending + immature + locked` (this is currently the case, but with locked coins being included in `available` already, they are double counted)

### Actual behavior
#598, while intending to clear up a separate calculation error, instead introduced another resulting in locked funds to be included as "Available"

### What version of PIVX Core are you using?
PIVX Core v3.1.1 distribution package

### Machine specs:
N/A (Applies to all)

### PR DESCRIPTION
**Pull request by Fuzzbawls**
Locked balances were being double counted in the total and available
values. This fixes the issue once and for all.

fixes #654 

`142bbc2` fixes the balance calculations
`0443512` improves code readability and show/hide conditions for watch-only balances
`45a1537` removes some trailing whitespace elsewhere in the file.

Keeping the commits separate to ease the review process.

### CODE REVIEW

File: src/qt/overviewpage.cpp [hunk 1]
```diff
@@ -267,30 +272,42 @@ void OverviewPage::setBalance(const CAmount& balance, const CAmount& unconfirmed
...
-    ui->labelBalance->setVisible(showPIVAvailable || showWatchOnlyPIVAvailable);
+
+    // PIV Available
+    bool showPIVAvailable = settingShowAllBalances || pivAvailableBalance != nTotalBalance;
+    bool showWatchOnlyPIVAvailable = settingShowAllBalances || nAvailableWatchBalance != nTotalWatchBalance;
     ui->labelBalanceText->setVisible(showPIVAvailable || showWatchOnlyPIVAvailable);
-    ui->labelWatchAvailable->setVisible(showPIVAvailable && showWatchOnly);
-    ui->labelUnconfirmed->setVisible(showPIVPending || showWatchOnlyPIVPending);
+    ui->labelBalance->setVisible(showPIVAvailable || showWatchOnlyPIVAvailable);
+    ui->labelWatchAvailable->setVisible(showWatchOnlyPIVAvailable && showWatchOnly);
```
Comment (Warrows) [inline src/qt/overviewpage.cpp]: This introduces a functional change I'm not sure we want.
See the difference:
**Before** the watchonly line was always shown, as long as the normal line is shown and there is at least one watchonly line:
capture d ecran de 2018-07-13 15-27-13
**After**, we would have empty lines:
capture d ecran de 2018-07-13 15-37-59

**Comment by Fuzzbawls** [comment]
I'll set aside some time this week to review the functional changes with watch-only addresses (who even uses them?)

**Comment by Mrs-X** [comment]
@Fuzzbawls : any news or should I merge?

**Comment by Fuzzbawls** [comment]
@Mrs-X still have to re-visit the noted concern with watch-only addresses, been out of town for a while.

**Pull request merged**